packages/core/src/StateMachine.ts

```diff
@@ -404,19 +428,12 @@ export class StateMachine<
   }
 
   public restoreState(
-    state: State<TContext, TEvent, TResolvedTypesMeta> | StateValue,
-    actorCtx?: ActorContext<TEvent, State<TContext, TEvent>>
+    state: State<TContext, TEvent, TResolvedTypesMeta> | StateValue
   ): State<TContext, TEvent, TResolvedTypesMeta> {
     const restoredState = isStateConfig(state)
       ? this.resolveState(state as any)
       : this.resolveState(State.from(state as any, this.context, this));
 
-    if (actorCtx) {
-      for (const action of restoredState.actions) {
-        execAction(action, restoredState, actorCtx);
-      }
-    }
```

Comment:
Wha's the current status of executing with restored states? What's the default expected result? Should entry actions be called? Should actions contained in `restoredState.actions` be restored?

Reply:
It depends, but we shouldn't _disallow_ actions from being re-executed, since this is a common expectation (client-side, not server-side).

Reply:
I'm OK with allowing that - it's just not clear to me what we are picking as default and how one can opt-into the non-default behavior.